src/class-wc-gateway-btcpay.php

```diff
@@ -756,6 +811,11 @@ public function process_payment($order_id)
             $invoice->setFullNotifications(true);
             $invoice->setExtendedNotifications(true);
 
+            // Handle additional tokens and enforce them for invoice payment.
+            if ($this->is_additional_token_method()) {
```

Comment:
Good point so you mean I should remove the if-statement completely and just setPaymentCurrencies with `null` (the $token_symbol property only exists on additional token classes) like `$invoice->setPaymentCurrencies(array($this->token_symbol));`? But this will lead to a "Undefined property" notice. 

But maybe you meant to just do a `if ($this->token_symbol) {`, this will work without notice and makes much sense. I can remove that helper method then.